be/src/exec/vectorized/iceberg/iceberg_delete_builder.h

```diff
@@ -0,0 +1,78 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#pragma once
+
+#include "common/status.h"
+#include "exec/vectorized/hdfs_scanner.h"
+
+namespace starrocks::vectorized {
+class PositionDeleteBuilder {
+public:
+    PositionDeleteBuilder() = default;
+    virtual ~PositionDeleteBuilder() = default;
+
+    virtual Status build(const std::string& timezone, std::set<std::int64_t> need_skip_rowids,
+                         const std::string& file_path, int64_t file_length) = 0;
+};
+
+class EqualityDeleteBuilder {
+public:
+    EqualityDeleteBuilder() = default;
+    virtual ~EqualityDeleteBuilder() = default;
+
+    virtual Status build(const std::string& timezone,
+                         std::unordered_map<SlotId, std::vector<ExprContext*>> conjunct_ctxs_by_slot,
+                         const std::string& file_path, int64_t file_length) = 0;
+};
+
+class ParquetPositionDeleteBuilder final : public PositionDeleteBuilder {
+public:
+    ParquetPositionDeleteBuilder(const HdfsScannerParams& scanner_params) : _scanner_params(scanner_params) {}
```

Comment:
plz don't pass `HdfsScannerParams`, every component should keep least knowledge priciple